Client/lib/gMap.js

```diff
@@ -0,0 +1,282 @@
+
+  var gMap = {}
+  
+  gMap.map;
+  gMap.markers = [];
+  
+  gMap.distance = [];
+  gMap.duration = [];
+  
+  gMap.pathLatLng = [];
+  gMap.travelMode = 'WALKING';
+  gMap.travelButtonList = [];
+
+  gMap.directionsDisplay;
+  gMap.directionsService;
+
+  gMap.startGMap = function (pos){
+    //can be -> BICYCLING, DRIVING, TRANSIT, WALKING
+    //----------------------------------
+    //creatig the Map
```

Comment:
You should consider breaking each of these labeled sections into independent functions.